File: mario.c
/*
Mario
Pseudocode:
Get integer to collect height from user
    Reject input if n<=0 or n>8
Print spaces and call hashtag function
Print gap
Call hashtag function for right pyramid
Print line to move onto next line
Ryan Ho
6/10/2022
*/

#include <cs50.h>
#include <stdio.h>

void hash(int line);
int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0 || height > 8);

    for (int line = 1; line <= height; line++) // Changing line value after completion of each line, stopping until line reaches height input
    {
        for (int space = 1; space <= height - line; space++) // For loop to determine number of spaces (height - line number = number of spaces needed)
        {
            printf(" ");
        }
        hash(line);
        printf("  "); // Print two spaces after right align hashtags
        hash(line);
        printf("\n");
    }

}

// Function
void hash(int line) // Void because nothing to return
{
    for (int digit = 1; digit <= line; digit++) // For loop to print hashtags needed for each step / columns with left alignment
    {
        printf("#");
    }
}

File: mario.py
# Objective: to recreate Mario in Python.
# Pseudocode in remark after each line
# Ryan Ho 2/7/23

import cs50

flag= True #flag for determining if height qualifies
while flag: #repeat until flag is switched to false
    height = int(input("Height: ")) #input is always a string, so need to do int() to typecast
    x = 1
    if height < 9 and height > 0:
        flag = False #switch flag to false if requirements fulfilled

for i in range(height): #for "height" number of rows
    for p in range(height-i-1): #For loop to print spaces
        print("  ", end = '') #Print two spaces without new line
    for j in range(i+1): #Print hashtags
        print("#", end = '')
    print(" ", end='') #Print space between hashtag
    for k in range(i+1): #print remaining hashtag
        print("#", end = '')
    print() #print newline
